network/wsPeer.go

```diff
@@ -534,11 +551,30 @@ func (wp *wsPeer) handleFilterMessage(msg IncomingMessage) {
 	}
 	var digest crypto.Digest
 	copy(digest[:], msg.Data)
-	//wp.net.log.Debugf("add filter %v", digest)
+	wp.net.log.Debugf("add filter %v", digest)
 	wp.outgoingMsgFilter.CheckDigest(digest, true, true)
 }
 
-func (wp *wsPeer) writeLoopSend(msg sendMessage) disconnectReason {
+var emptyHash = crypto.Digest{}
+
+func (wp *wsPeer) writeLoopSend(msgs sendMessages) disconnectReason {
+	for i, msg := range msgs.msgs {
+		select {
+		case <-msg.ctx.Done():
+			logging.Base().Infof("cancelled large send, msg %v out of %v", i, len(msgs.msgs))
```

Comment:
Avoid logging messages in the network library. Enable these while you're working on a concrete test. 
( it's ok to leave these commented out in the code for that purpose )